File: tools/consumer-smoke/templates/dart/smoke.dart
// 게시본을 pub 캐시에서 그대로 import 한다 — 경로 의존(path:)이 하나도 없어야 한다.
import 'dart:convert';
import 'dart:io';

import 'package:rynl10n/rynl10n.dart';

void main() {
  final bundle = parseBakedSnapshot(File('snapshot.json').readAsStringSync(),
      source: 'snapshot.json');
  final client = RynL10nClient(
    bundle: bundle,
    store: InMemoryDeliveryStore(),
    context: ClientContext(appVersion: '3.2.1'),
    locale: 'en',
  );

  var bad = 0;
  final checks = jsonDecode(File('checks.json').readAsStringSync()) as List<dynamic>;
  for (final raw in checks) {
    final c = raw as Map<String, dynamic>;
    final got = client.t(
      c['key'] as String,
      args: (c['args'] as Map<String, dynamic>).cast<String, Object?>(),
      locale: c['locale'] as String?,
    );
    final ok = got == c['expect'];
    if (!ok) bad++;
    stdout.writeln('${ok ? "PASS" : "FAIL"}  ${c['name']}: "$got"'
        '${ok ? "" : ' (기대 "${c['expect']}")'}');
  }
  exitCode = bad == 0 ? 0 : 1;
}

File: tools/consumer-smoke/templates/web/smoke.mjs
// 게시본을 node_modules에서 그대로 import 한다 — 저장소 소스를 가리키는 경로가 하나도 없어야 한다.
import { HttpRynL10n, BakedBundle } from "@rynl10n/web";
import { readFileSync } from "node:fs";

const read = (f) => JSON.parse(readFileSync(new URL(f, import.meta.url), "utf8"));
const sdk = new HttpRynL10n({
  projectKey: "smoke",
  endpoint: "https://cdn.example.invalid", // 배포 플레인 — 스모크는 네트워크를 타지 않는다
  bundle: BakedBundle.parse(read("./snapshot.json")),
  context: { appVersion: "3.2.1" },
  locale: "en",
});

let bad = 0;
for (const c of read("./checks.json")) {
  const got = sdk.t(c.key, c.args, c.locale ?? undefined);
  const ok = got === c.expect;
  if (!ok) bad++;
  console.log(`${ok ? "PASS" : "FAIL"}  ${c.name}: ${JSON.stringify(got)}${ok ? "" : ` (기대 ${JSON.stringify(c.expect)})`}`);
}
process.exit(bad === 0 ? 0 : 1);
